packages/react/src/xcss-prop/index.ts

```diff
@@ -0,0 +1,141 @@
+import type * as CSS from 'csstype';
+
+import { ac } from '../runtime';
+import type { CSSPseudos } from '../types';
+
+type CSSProperties = Readonly<CSS.PropertiesFallback<number | string>>;
+
+type CSSPseudoRule = { [Q in CSSPseudos]?: CSSProperties };
+
+type XCSSItem<TStyleDecl extends keyof CSSProperties> = {
+  [Q in keyof CSSProperties]: Q extends TStyleDecl
+    ? CompiledPropertyDeclarationReference | string | number
+    : never;
+};
+
+type XCSSPseudos<K extends keyof CSSProperties, TPseudos extends CSSPseudos> = {
+  [Q in CSSPseudos]?: Q extends TPseudos ? XCSSItem<K> : never;
+};
+
+/**
+ * We currently block all at rules from xcss prop.
+ * This needs us to decide on what the final API is across Compiled to be able to set.
+ */
+type XCSSAtRules = {
+  [Q in CSS.AtRules]?: never;
+};
+
+export type CSSRuleDefinition = CSSProperties & CSSPseudoRule;
+
+declare const __classref: unique symbol;
```

Comment:
One thought: This is going to be referentially equivalent on both ends. In a monorepo world this is OK. In a platform vs. ecosystem world where Compiled might be outdated in Ecosystem 2p/3p but latest in platform it will be a problem as they could be on two different major versions (our more likely right now, two different minor versions pre 1.0!).

I'll see if I can move this to a string instead.